colossalai/auto_parallel/solver/cost_graph.py

```diff
@@ -27,6 +29,15 @@ def __init__(self, leaf_strategies, simplify=True):
         self.simplify = simplify
         self._build_cost_graph()
 
+    def _remove_invalid_node(self, node, attr_name):
+        remove_list = []
+        target_node_list = getattr(node, attr_name, None)
+        for target_node in target_node_list:
+            if target_node not in self.nodes:
+                remove_list.append(target_node)
+        for element in remove_list:
+            target_node_list.remove(element)
```

Comment:
If this node does not have `attr_name`, `target_node_list` will be None, and None cannot be iterated.